src/main/java/rtg/event/EventManagerRTG.java

```diff
@@ -107,22 +104,27 @@ public void generateMinableRTG(OreGenEvent.GenerateMinable event) {
                     return;
 
                 default:
-                	return;
+                	break;
             }
         }
     }
 
     public class InitBiomeGensRTG
     {
-        InitBiomeGensRTG()
-        {
-            Logger.debug(EVENT_SYSTEM + "Initialising InitBiomeGensRTG");
+        InitBiomeGensRTG() {
+            logEventMessage("Initialising InitBiomeGensRTG...");
         }
 
         @SubscribeEvent
         public void initBiomeGensRTG(WorldTypeEvent.InitBiomeGens event) {
 
-            if (event.newBiomeGens[0].getClass().getName().contains("GenLayerEB")) return;
+            if (!(event.worldType instanceof WorldTypeRTG)) {
+                return;
```

Comment:
I don't think any of RTG's other event handlers need to be unregistered this early, as they'll all get unregistered before the non-RTG world loads (WorldEvent.Load), but... like I said in the comments, it's better to be safe than sorry, so unregistering them here as well doesn't hurt anything.